### PR TITLE
Convert all docs/*.html to .md

### PR DESCRIPTION
**Pull request by levitte**
We only take the easy parts, and leave manual page generation files alone.

Partial fix of #324

### CODE REVIEW

**Review by paulidale — APPROVED** [review]

**Comment by levitte** [comment]
Actually, I had a thought with this one that will need re-approval later

File: docs/fips.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+## Background
+
+Please please read the [User Guide](fips/UserGuide.pdf).
+
+-   OpenSSL itself is not validated. Instead a special carefully defined
+    software component called the OpenSSL FIPS Object Module has been
+    created. This Module was designed for compatibility with OpenSSL so
+    that products using the OpenSSL API can be converted to use
+    validated cryptography with minimal effort.
+-   The OpenSSL FIPS Object Module 2.0 validation is \"delivered\" in
```
Comment (paulidale) [inline docs/fips.md]: Spurious \\ here and later

**Review by paulidale — CHANGES_REQUESTED** [review]

File: docs/fips.md (hunk 1 above)
Comment (levitte) [reply docs/fips.md]: Fixed most of them

**Review by paulidale — APPROVED** [review]

**Comment by levitte** [comment]
There were a few more files to fix...  Re-approval needed

**Review by paulidale — DISMISSED** [review]

**Comment by levitte** [comment]
I noticed this work wasn't quite complete, so back to draft

**Comment by levitte** [comment]
Rebased ~~and squashed~~

**Comment by levitte** [comment]
Squashed and generally cleaned up

**Comment by levitte** [comment]
https://beta.openssl.org/ is the test bed for this

**Comment by t8m** [comment]
I see some _interesting_ files in the directory list at: https://beta.openssl.org/docs/fips.html
- the .md files and dirdata.yaml in particular. Can we somehow filter them out?

**Comment by levitte** [comment]
> I see some _interesting_ files in the directory list at: https://beta.openssl.org/docs/fips.html
> 
>     * the .md files and dirdata.yaml in particular. Can we somehow filter them out?

Yes, see ~~568c8e4fa30deb06b83cacb2924ec60f19515ff1~~27d41d0f2891943c27692007f8e70face3e5c3c7

**Review by t8m — COMMENTED** [review]
Now the only glitch I see in docs on beta.openssl.org is that there are manual pages only for the master branch. The rest is giving a weird 404 page. For example: https://beta.openssl.org/docs/man3.0

**Comment by levitte** [comment]
Ping

**Comment by t8m** [comment]
@levitte Is my comment above relevant?

**Comment by levitte** [comment]
> Now the only glitch I see in docs on beta.openssl.org is that there are manual pages only for the master branch. The rest is giving a weird 404 page. For example: https://beta.openssl.org/docs/man3.0

The default `make` command only builds the master docs...  I'm running `make docs` now, which builds the whole set...  so er, in 15-20 minutes?

**Comment by levitte** [comment]
> > Now the only glitch I see in docs on beta.openssl.org is that there are manual pages only for the master branch. The rest is giving a weird 404 page. For example: https://beta.openssl.org/docs/man3.0
> 
> The default `make` command only builds the master docs... I'm running `make docs` now, which builds the whole set... so er, in 15-20 minutes?

Done

**Comment by t8m** [comment]
LGTM

**Comment by levitte** [comment]
> LGTM

@openssl/omc, final review?

**Review by mattcaswell — APPROVED** [review]

**Comment by levitte** [comment]
Merged

9d6ac3f1db833a495df684429215e31c76c8ea4e Convert docs/*.html to .md
964dd4386ddb005b1dc52eb197ac53f4686041b2 Convert docs/sub-*.html.tt to .md.tt
18aaf09b6e2dc3b4e1194d561059cec9a2339155 bin/from-tt: Correct bugs for determining [% top %]
e0824ffd90afab8425e01c90594c03b13d9d9975 Bugfix: make knows $(dir ...), not $(dirname ...)

**Pull request closed without merging**